runtime/array.c

```diff
@@ -346,6 +335,25 @@ CAMLprim value caml_make_vect(value len, value init)
   CAMLreturn (res);
 }
 
+#ifndef FLAT_FLOAT_ARRAY
+static value uninitialized_float = Val_unit;
```

Comment:
`uninitialized_float` could be moved into `caml_make_float_vect` to reduce its scope, and to avoid the need for two sets of `#ifdef` conditions.

Reply:
Done.